### PR TITLE
wait

### PR DESCRIPTION
**Pull request by Supermegadex**
Hey, there!

I needed something just like your project for a presentation, but I wanted to be able to wait a little while between instructions, so I made an instruction that adds some delay. I hope this change is helpful!

-Daniel

### CODE REVIEW

File: src/instructions.ts [hunk 1]
```diff
@@ -6,15 +6,17 @@ export type Command = {
   args: any[];
 };
 
-export type TypeText = { type: 'typeText'; text: string[] };
+export type Wait = { type: 'wait', length: number | 'manual' };
```
Comment (mauricedb) [inline src/instructions.ts]: Can you change `length` to `delay` here so it is the same as in the `TypeText` command please?

File: src/instruction-handlers.ts [hunk 2]
```diff
@@ -79,9 +80,27 @@ export const goto = async (instruction: GoTo): Promise<void> => {
...
-  await wait(getPause());
+  await timeout(getPause());
 };
 
-function wait(time: number) {
+export const wait = (instruction: Wait): Promise<void> => {
+  return new Promise(async (resolve, reject) => {
+    if (typeof instruction.length === 'number') {
+      await timeout(instruction.length);
+      await timeout(getPause());
```
Comment (mauricedb) [inline src/instruction-handlers.ts]: Why the `await timeout(getPause());`? Seems the first await should be enough as that is what is configured.

**Comment by mauricedb** [comment]
Hi @Supermegadex 	

👍 	Great feature, this was something I was already planning but you did in a better way 😄 	

One thing, can you remove the `package-lock.json` as I am using Yarn here for the dependencies?

I left a few other review items.

Thanks

**Comment by Supermegadex** [comment]
I made the changes you requested. Sorry I didn't get to it sooner--I didn't notice that you had made these comments because I am new to contributing and I assumed GitHub would send me a notification. Let me know if you need me to make any more changes or if you need help with something else!

-Daniel

**Comment by mauricedb** [comment]
Hi @Supermegadex,

No problem, with GIT you need to watch a repository to be notified of updates.

I have merged the PR and published version 0.2.0 that should be available any minute now at https://marketplace.visualstudio.com/items?itemName=mauricedebeijer.presentation-buddy

Thanks for the PR
Maurice

**Pull request merged**